src/utils/FocusList.ts

```diff
@@ -0,0 +1,114 @@
+export class FocusList<T> {
+    private readonly array: T[];
+    private focus: number;
+
+    constructor(array: T[] = [], focus: number = 0) {
+        this.array = array
+        this.focus = focus
+    }
+
+    length() {
+        return this.array.length;
+    }
+
+    getFocus() {
+        return this.focus
+    }
+
+    setFocus(focus: number) {
+        if(focus >= this.length())
+        this.focus = focus
+    }
+
+    getContent() {
+        return this.array;
+    }
+
+    getItemInFocus() {
+        return this.array[this.focus]
+    }
+
+    shiftFocusLeft() {
+        if (this.focus === 0) this.focus = this.length() - 1
+        else this.focus = this.focus - 1
+    }
+
+    shiftFocusRight() {
+        if (this.focus === this.length() - 1) this.focus = 0
+        else this.focus = this.focus + 1
+    }
+
+    contains(item: T){
+        return this.array.includes(item)
+    }
+
+    push(items: T[]) {
+        const itemsIncluded = items.filter(item => this.contains(item))
+        if (itemsIncluded.length > 0) throw new Error("There are " + itemsIncluded.length
+            + " items already included in this list. "
+            + "Prevent repetition of items: " + itemsIncluded.join() + ".")
+        else this.array.push(...items)
+    }
+
+    delete(item: T) {
+        const index = this.array.indexOf(item);
+        if (index === undefined) throw new Error("There is no item " + item + " to delete in this list.")
```

Comment:
```suggestion
        if (!index) throw new Error("There is no item " + item + " to delete in this list.")
```
or do you need to check for just and only undefined?